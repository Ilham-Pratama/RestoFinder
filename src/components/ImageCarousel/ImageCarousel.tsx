import React from 'react';
import { FlatList, Image, StyleSheet, useWindowDimensions } from 'react-native';

interface ImageCarouselProps {
  images: string[];
}

const ImageCarousel = ({ images }: ImageCarouselProps) => {
  const { width } = useWindowDimensions();

  return (
    <FlatList
      accessibilityRole="list"
      data={images}
      keyExtractor={(image, idx) => `${image} - ${idx}`}
      renderItem={({ item }) => (
        <Image
          accessibilityRole="image"
          source={{ uri: item }}
          style={{ width }}
        />
      )}
      horizontal
      pagingEnabled
      showsHorizontalScrollIndicator={false}
      bounces={false}
      style={[styles.listContainer]}
    />
  );
};

const styles = StyleSheet.create({
  listContainer: {
    height: 250,
    overflow: 'hidden'
  }
});

export default ImageCarousel;
