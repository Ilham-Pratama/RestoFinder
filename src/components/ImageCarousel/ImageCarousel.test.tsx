import { render } from '@testing-library/react-native';
import React from 'react';
import ImageCarousel from '.';

const WIDTH = 120;
const IMAGES = ['image-1', 'image-2', 'image-3'];

jest.mock('react-native', () => {
  return {
    FlatList: jest.requireActual('react-native').FlatList,
    Image: jest.requireActual('react-native').Image,
    StyleSheet: jest.requireActual('react-native').StyleSheet,
    useWindowDimensions: () => ({ width: WIDTH })
  };
});

describe('ImageCarousel', () => {
  it('renders ImageCarousel', () => {
    const component = render(<ImageCarousel images={IMAGES} />);

    expect(component.toJSON()).toMatchSnapshot();

    const flatList = component.getByRole('list');

    expect(flatList.props.data).toHaveLength(IMAGES.length);
    expect(component.getAllByRole('image')).toHaveLength(IMAGES.length);
    expect(component.getAllByRole('image')[0].props).toHaveProperty(
      'source.uri',
      IMAGES[0]
    );
  });
});
