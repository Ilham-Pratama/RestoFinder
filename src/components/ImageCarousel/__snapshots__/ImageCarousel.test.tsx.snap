// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCarousel renders ImageCarousel 1`] = `
<RCTScrollView
  accessibilityRole="list"
  bounces={false}
  data={
    Array [
      "image-1",
      "image-2",
      "image-3",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  pagingEnabled={true}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Array [
      Object {
        "height": 250,
        "overflow": "hidden",
      },
    ]
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <Image
        accessibilityRole="image"
        source={
          Object {
            "uri": "image-1",
          }
        }
        style={
          Object {
            "width": 120,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <Image
        accessibilityRole="image"
        source={
          Object {
            "uri": "image-2",
          }
        }
        style={
          Object {
            "width": 120,
          }
        }
      />
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <Image
        accessibilityRole="image"
        source={
          Object {
            "uri": "image-3",
          }
        }
        style={
          Object {
            "width": 120,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
